rules/DowngradePhp70/Rector/FunctionLike/DowngradeThrowableTypeDeclarationRector.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\FunctionLike;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ArrowFunction;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Function_;
+use PHPStan\Type\ObjectType;
+use Rector\BetterPhpDocParser\PhpDocParser\PhpDocFromTypeDeclarationDecorator;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\DowngradePhp70\Rector\FunctionLike\DowngradeThrowableTypeDeclarationRector\DowngradeThrowableTypeDeclarationRectorTest
+ */
+final class DowngradeThrowableTypeDeclarationRector extends AbstractRector
+{
+    public function __construct(
+        private readonly PhpDocFromTypeDeclarationDecorator $phpDocFromTypeDeclarationDecorator,
+    ) {
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Function_::class, ClassMethod::class, Closure::class, ArrowFunction::class];
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Remove `Throwable` return type, add a `@return Throwable` tag instead',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function foo(): \Throwable
+    {
+        return new \Exception("Troubles");
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function foo()
+    {
+        return new \Exception("Troubles");
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param ClassMethod|Function_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $throwableType = new ObjectType('Throwable');
+
+        foreach ($node->getParams() as $param) {
+            $this->phpDocFromTypeDeclarationDecorator->decorateParamWithSpecificType($param, $node, $throwableType);
```

Comment:
Currently does not seem to work because `Rector\StaticTypeMapper\ValueObject\Type\FullyQualifiedObjectType` is not `PHPStan\Type\ObjectType` as expected by the comparison:

https://github.com/rectorphp/rector-src/blob/eac5313f26a57e418ef66e55028319b339065d49/packages/BetterPhpDocParser/PhpDocParser/PhpDocFromTypeDeclarationDecorator.php#L186